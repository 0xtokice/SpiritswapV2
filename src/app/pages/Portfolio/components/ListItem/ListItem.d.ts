export interface Props {
  tokenName: string;
  tokenAmount: string | number;
  tokenAddress?: string;
  usdAmount: string | number;
  options: { id: number | string; value: string; type: string }[];
}
