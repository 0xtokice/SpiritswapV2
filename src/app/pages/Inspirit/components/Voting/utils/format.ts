export const formatVotes = (base: number) =>
  `${`${(base / 1000000).toFixed(1)}`}M`;
