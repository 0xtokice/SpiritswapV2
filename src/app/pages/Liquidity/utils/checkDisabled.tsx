const checkDisabled = (firstToken, secondToken, balanceOne, balanceTwo) => {
  return (
    firstToken.value === '' ||
    firstToken.value === '0' ||
    firstToken.value === '0.0' ||
    secondToken.value === '' ||
    secondToken.value === '0' ||
    secondToken.value === '0.0' ||
    balanceOne === 0 ||
    balanceTwo === 0 ||
    isNaN(firstToken.value) ||
    isNaN(secondToken.value) ||
    isNaN(balanceOne) ||
    isNaN(balanceTwo)
  );
};

export default checkDisabled;
