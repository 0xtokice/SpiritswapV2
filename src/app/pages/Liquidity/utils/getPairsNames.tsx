const getPairsNames = (pairsList: string[]) => {
  const arrayLength = pairsList?.length;
  let finalString = '';
  const isTheLast = arrayLength - 1;
  pairsList?.forEach((item, index) => {
    if (isTheLast === index) return (finalString += item);

    finalString += item + ' + ';
  });
  return finalString;
};

export default getPairsNames;
