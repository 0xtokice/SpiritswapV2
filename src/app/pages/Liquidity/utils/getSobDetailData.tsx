import { tokenData } from 'utils/data/types';
import { LiquidityDetailProps } from './getDetailData';

const getSobDetailData = (pair: tokenData) => {
  let data: LiquidityDetailProps[] = [];

  const { lpSupply, amount, tokensAmounts } = pair;

  const poolTokenPercentage = lpSupply
    ? `${((parseFloat(amount) / lpSupply) * 100).toFixed(5)} %`
    : 'Unknown'; // get the share pool percentaje

  tokensAmounts?.forEach(tokenWAmount => {
    data.push({
      detailTitle: `Pooled ${tokenWAmount.token.symbol}:`,
      detailValue: tokenWAmount.amount ? `${tokenWAmount.amount}` : 'Unknown',
    });
  });

  data = [
    ...data,
    {
      detailTitle: 'Your pool share:',
      detailValue: `${poolTokenPercentage}`,
    },
  ];

  data = data.map((item, i) => {
    return { ...item, id: i };
  });

  return data;
};

export default getSobDetailData;
