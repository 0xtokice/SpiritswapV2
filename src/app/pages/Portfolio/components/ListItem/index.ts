export type { Props as ListItemProps } from './ListItem.d';
export { default as ListItem } from './ListItem';
