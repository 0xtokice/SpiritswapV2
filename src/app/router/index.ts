export { default as SiteRouting } from './SiteRouting';
export { default as RootPathContext } from './RootPathContext';
