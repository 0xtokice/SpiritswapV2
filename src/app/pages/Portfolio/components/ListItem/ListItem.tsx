import { Props } from './ListItem.d';
import { StyledContainer } from './styles';
import ImageLogo from 'app/components/ImageLogo';
import { Flex, Text } from '@chakra-ui/react';
import { NewDropdown } from 'app/components/Menu';
import { truncateTokenValue } from 'app/utils';

const ListItem = ({
  tokenName,
  tokenAddress,
  tokenAmount,
  usdAmount,
  options,
  ...props
}: Props) => {
  const tokenArray = Array.isArray(tokenName) ? tokenName : [tokenName];
  const isLP = tokenArray[0] ? tokenArray[0].includes('/') : false;
  const tokenSymbol = tokenArray[0]
    ? tokenArray[0].replace(' LP', '').trimEnd()
    : '';
  const tokenSymbols = isLP && tokenSymbol.split('/');

  return (
    <StyledContainer data-testid="ListItem" {...props}>
      <Flex align="center">
        <Flex w={isLP ? '55px' : '40px'}>
          {tokenSymbols ? (
            tokenSymbols.map((symbol, index) => (
              <ImageLogo
                size="32px"
                nextPair={index > 0 ?? false}
                symbol={symbol}
                key={`token-icon-${symbol}`}
              />
            ))
          ) : (
            <ImageLogo
              size="32px"
              symbol={tokenSymbol}
              key={`token-icon-${tokenSymbol}`}
            />
          )}
        </Flex>
        <Text fontWeight="medium">{tokenArray.join('/')}</Text>
      </Flex>
      <Flex align="center">
        <Flex direction="column" align="flex-end">
          <Text>
            {truncateTokenValue(
              Number(tokenAmount),
              Number(usdAmount) / Number(tokenAmount),
            )}
          </Text>
          <Text fontSize="sm" color="grayDarker">
            {`$${truncateTokenValue(Number(usdAmount))}`}
          </Text>
        </Flex>
        <NewDropdown items={options} address={tokenAddress ?? '0x0000000'} />
      </Flex>
    </StyledContainer>
  );
};

export default ListItem;
