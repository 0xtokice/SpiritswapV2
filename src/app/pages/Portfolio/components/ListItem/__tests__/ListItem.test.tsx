import React from 'react';
import ReactDOM from 'react-dom';
import ListItem from '../ListItem';
import { Props as ListItemProps } from '../ListItem.d';
import { render, cleanup } from '@testing-library/react';
import ThemeProvider from 'theme';
import renderer from 'react-test-renderer';
import { act } from 'react-dom/test-utils';
import { Token } from 'app/utils';
import { TokenOptions } from 'app/utils/tokenOptions';
import { ChakraThemeWrapper } from 'app/components/shared/testing/ChakraTheme';
import { SPIRIT } from 'constants/index';

describe('ListItem-Component', () => {
  let container;
  const ListItemTestId = 'ListItem';
  const { act: rendererAct, create } = renderer;
  const ListItemComponent = ({ ...props }: ListItemProps) => {
    return (
      <ChakraThemeWrapper>
        <ThemeProvider>
          <React.StrictMode>
            <ListItem {...props} data-testid={ListItemTestId} />
          </React.StrictMode>
        </ThemeProvider>
      </ChakraThemeWrapper>
    );
  };

  beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    document.body.removeChild(container);
    container = null;
    cleanup();
  });

  it('renders without crashing and without empty values ', () => {
    act(() => {
      ReactDOM.render(
        <ListItemComponent
          tokenName=""
          tokenAmount=""
          usdAmount=""
          options={TokenOptions(SPIRIT.address)}
        />,
        container,
      );
    });
  });

  it('renders without crashing and with Values', () => {
    act(() => {
      ReactDOM.render(
        <ListItemComponent
          tokenName={Token.DAI}
          tokenAmount="4,050.33"
          usdAmount="5,530.20"
          options={TokenOptions(SPIRIT.address)}
        />,
        container,
      );
    });
    act(() => {
      ReactDOM.render(
        <ListItemComponent
          tokenName={Token.DAI}
          tokenAmount="4,050.33"
          usdAmount="5,530.20"
          options={TokenOptions(SPIRIT.address)}
        />,
        container,
      );
    });
  });

  xit('renders ListItem with All Props', () => {
    const tokenName = Token.DAI;
    const tokenAmount = '4,050.33';
    const usdAmount = '5,530.20';
    const options = TokenOptions;
    const { getByTestId } = render(
      <ListItemComponent
        tokenName={tokenName}
        tokenAmount={tokenAmount}
        usdAmount={usdAmount}
        options={options(SPIRIT.address)}
      />,
    );

    expect(getByTestId(ListItemTestId)).toHaveTextContent(tokenName);
    expect(getByTestId(ListItemTestId)).toHaveTextContent(tokenAmount);
    expect(getByTestId(ListItemTestId)).toHaveTextContent(usdAmount);
  });

  it('OnClick', () => {
    const { container } = render(
      <ListItemComponent
        tokenName={Token.DAI}
        tokenAmount="4,050.33"
        usdAmount="5,530.20"
        options={TokenOptions(SPIRIT.address)}
      />,
    );
    const Svg = container.querySelector('svg');
    expect(Svg).toBeTruthy();
  });

  it('Matches SnapShot with Empty Values', () => {
    let ListItemSnapShot;
    rendererAct(() => {
      ListItemSnapShot = create(
        <ListItemComponent
          tokenName=""
          tokenAmount={''}
          usdAmount={''}
          options={TokenOptions(SPIRIT.address)}
        />,
      );
    });

    expect(ListItemSnapShot.toJSON()).toMatchSnapshot();
  });

  it('Matches SnapShot with Values and singal Token', () => {
    let ListItemSnapShot;
    const tokenName = Token.DAI;
    const tokenAmount = '4,050.33';
    const usdAmount = '5,530.20';
    const options = TokenOptions;
    rendererAct(() => {
      ListItemSnapShot = create(
        <ListItemComponent
          tokenName={tokenName}
          tokenAmount={tokenAmount}
          usdAmount={usdAmount}
          options={options(SPIRIT.address)}
        />,
      );
    });

    expect(ListItemSnapShot.toJSON()).toMatchSnapshot();
  });

  it('Matches SnapShot with Values with Double Token', () => {
    let ListItemSnapShot;
    const tokenName = Token.DAI;
    const tokenAMount = '4,050.33';
    const usdAmount = '5,530.20';
    const options = TokenOptions;
    rendererAct(() => {
      ListItemSnapShot = create(
        <ListItemComponent
          tokenName={tokenName}
          tokenAmount={tokenAMount}
          usdAmount={usdAmount}
          options={options(SPIRIT.address)}
        />,
      );
    });

    expect(ListItemSnapShot.toJSON()).toMatchSnapshot();
  });
});
