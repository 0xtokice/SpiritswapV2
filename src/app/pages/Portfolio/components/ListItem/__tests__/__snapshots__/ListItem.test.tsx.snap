// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem-Component Matches SnapShot with Empty Values 1`] = `
Array [
  <div
    className="css-15nwh8t"
    data-testid="ListItem"
  >
    <div
      className="css-70qvj9"
    >
      <div
        className="css-1e7o9jt"
      >
        <div
          className="css-k008qs"
        >
          <img
            alt=" logo"
            className="chakra-image__placeholder css-120tod0"
            fallbackStrategy="onError"
            src="/images/tokens/default.png"
          />
        </div>
      </div>
      <p
        className="chakra-text css-g4p9of"
      >
        
      </p>
    </div>
    <div
      className="css-70qvj9"
    >
      <div
        className="css-qfdd9t"
      >
        <p
          className="chakra-text css-0"
        >
          0.00
        </p>
        <p
          className="chakra-text css-15e15uw"
        >
          $0.00
        </p>
      </div>
      <button
        aria-controls="menu-list-4"
        aria-expanded={false}
        aria-haspopup="menu"
        aria-label="Options"
        className="chakra-button chakra-menu__menu-button css-129vlsj"
        id="menu-button-4"
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="chakra-icon css-69nngz"
          focusable={false}
        >
          dots-outline.svg
        </svg>
      </button>
      <div
        className="css-r6z5ec"
        style={
          Object {
            "inset": "0 auto auto 0",
            "minWidth": "max-content",
            "position": "absolute",
            "visibility": "hidden",
          }
        }
      >
        <div
          aria-orientation="vertical"
          className="chakra-menu__menu-list css-7zx23t"
          id="menu-list-4"
          onKeyDown={[Function]}
          role="menu"
          style={
            Object {
              "opacity": 0,
              "transform": "scale(0.8) translateZ(0)",
              "transformOrigin": "var(--popper-transform-origin)",
              "visibility": "hidden",
            }
          }
          tabIndex={-1}
        >
          <button
            className="chakra-menu__menuitem css-inosk3"
            id="menu-list-4-menuitem-1"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            role="menuitem"
            tabIndex={0}
            type="button"
          >
            Swap
          </button>
          <button
            className="chakra-menu__menuitem css-inosk3"
            id="menu-list-4-menuitem-2"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            role="menuitem"
            tabIndex={0}
            type="button"
          >
            View on Ftmscan
          </button>
        </div>
      </div>
    </div>
  </div>,
  <span />,
]
`;

exports[`ListItem-Component Matches SnapShot with Values and singal Token 1`] = `
Array [
  <div
    className="css-15nwh8t"
    data-testid="ListItem"
  >
    <div
      className="css-70qvj9"
    >
      <div
        className="css-1e7o9jt"
      >
        <div
          className="css-k008qs"
        >
          <img
            alt="DAI logo"
            className="chakra-image__placeholder css-120tod0"
            fallbackStrategy="onError"
            src="/images/tokens/default.png"
          />
        </div>
      </div>
      <p
        className="chakra-text css-g4p9of"
      >
        DAI
      </p>
    </div>
    <div
      className="css-70qvj9"
    >
      <div
        className="css-qfdd9t"
      >
        <p
          className="chakra-text css-0"
        >
          0.00
        </p>
        <p
          className="chakra-text css-15e15uw"
        >
          $0.00
        </p>
      </div>
      <button
        aria-controls="menu-list-4"
        aria-expanded={false}
        aria-haspopup="menu"
        aria-label="Options"
        className="chakra-button chakra-menu__menu-button css-129vlsj"
        id="menu-button-4"
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="chakra-icon css-69nngz"
          focusable={false}
        >
          dots-outline.svg
        </svg>
      </button>
      <div
        className="css-r6z5ec"
        style={
          Object {
            "inset": "0 auto auto 0",
            "minWidth": "max-content",
            "position": "absolute",
            "visibility": "hidden",
          }
        }
      >
        <div
          aria-orientation="vertical"
          className="chakra-menu__menu-list css-7zx23t"
          id="menu-list-4"
          onKeyDown={[Function]}
          role="menu"
          style={
            Object {
              "opacity": 0,
              "transform": "scale(0.8) translateZ(0)",
              "transformOrigin": "var(--popper-transform-origin)",
              "visibility": "hidden",
            }
          }
          tabIndex={-1}
        >
          <button
            className="chakra-menu__menuitem css-inosk3"
            id="menu-list-4-menuitem-1"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            role="menuitem"
            tabIndex={0}
            type="button"
          >
            Swap
          </button>
          <button
            className="chakra-menu__menuitem css-inosk3"
            id="menu-list-4-menuitem-2"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            role="menuitem"
            tabIndex={0}
            type="button"
          >
            View on Ftmscan
          </button>
        </div>
      </div>
    </div>
  </div>,
  <span />,
]
`;

exports[`ListItem-Component Matches SnapShot with Values with Double Token 1`] = `
Array [
  <div
    className="css-15nwh8t"
    data-testid="ListItem"
  >
    <div
      className="css-70qvj9"
    >
      <div
        className="css-1e7o9jt"
      >
        <div
          className="css-k008qs"
        >
          <img
            alt="DAI logo"
            className="chakra-image__placeholder css-120tod0"
            fallbackStrategy="onError"
            src="/images/tokens/default.png"
          />
        </div>
      </div>
      <p
        className="chakra-text css-g4p9of"
      >
        DAI
      </p>
    </div>
    <div
      className="css-70qvj9"
    >
      <div
        className="css-qfdd9t"
      >
        <p
          className="chakra-text css-0"
        >
          0.00
        </p>
        <p
          className="chakra-text css-15e15uw"
        >
          $0.00
        </p>
      </div>
      <button
        aria-controls="menu-list-4"
        aria-expanded={false}
        aria-haspopup="menu"
        aria-label="Options"
        className="chakra-button chakra-menu__menu-button css-129vlsj"
        id="menu-button-4"
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="chakra-icon css-69nngz"
          focusable={false}
        >
          dots-outline.svg
        </svg>
      </button>
      <div
        className="css-r6z5ec"
        style={
          Object {
            "inset": "0 auto auto 0",
            "minWidth": "max-content",
            "position": "absolute",
            "visibility": "hidden",
          }
        }
      >
        <div
          aria-orientation="vertical"
          className="chakra-menu__menu-list css-7zx23t"
          id="menu-list-4"
          onKeyDown={[Function]}
          role="menu"
          style={
            Object {
              "opacity": 0,
              "transform": "scale(0.8) translateZ(0)",
              "transformOrigin": "var(--popper-transform-origin)",
              "visibility": "hidden",
            }
          }
          tabIndex={-1}
        >
          <button
            className="chakra-menu__menuitem css-inosk3"
            id="menu-list-4-menuitem-1"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            role="menuitem"
            tabIndex={0}
            type="button"
          >
            Swap
          </button>
          <button
            className="chakra-menu__menuitem css-inosk3"
            id="menu-list-4-menuitem-2"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            role="menuitem"
            tabIndex={0}
            type="button"
          >
            View on Ftmscan
          </button>
        </div>
      </div>
    </div>
  </div>,
  <span />,
]
`;
